lib/bazel/options.go

```diff
@@ -0,0 +1,111 @@
+package bazel
+
+import (
+	"os/exec"
+)
+
+// subcommand is implemented by an arguments struct for each bazel subcommand
+// (such as 'query', 'build', etc.).
+type subcommand interface {
+	// Args returns the subcommand plus all the subcommands' arguments.
+	Args() []string
+}
+
+// baseOptions captures supported bazel startup flags.
+type baseOptions struct {
+	// Bazel's cache directory for this workspace.
+	OutputBase string
+}
+
+// Option modifies Bazel startup options.
+type Option func(*baseOptions)
+
+// WithOutputBase sets --output_base for this bazel invocation.
+func WithOutputBase(outputBase string) Option {
+	return func(o *baseOptions) {
+		o.OutputBase = outputBase
+	}
+}
+
+// flags returns the startup flags as passed to bazel.
+func (o *baseOptions) flags() []string {
+	var f []string
+	if o.OutputBase != "" {
+		f = append(f, "--output_base", o.OutputBase)
+	}
+	return f
+}
+
+// apply applies all the options to this option struct.
```

Comment:
lol. Generally in the repository we've used the opposite convention, see example below.

I would stick to what we did in the rest of the repository, not because I have any opinion on one being worse/better than the other, but for consistency. It took me 30 seconds to realize what was different here.

```
type Options []Option

func (opts Options) apply(o *baseOptions) {
  for _, opt := range opts {
    opt(o)
  }
}
```

Reply:
Fixed